### PR TITLE
Implement refresh token handling in frontend

### PR DESCRIPTION
**Pull request by iiNomad23**

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -28,12 +29,12 @@ const router = createBrowserRouter([
     },
 ]);
 
-function App() {
+const App = (): React.ReactElement => {
```
Comment (Smightym8) [inline src/App.tsx:32]: What does `React.ReactElement` change compared to how it was before?

File: src/hooks/services/useAuthenticationService.ts [hunk 2]
```diff
@@ -19,7 +19,7 @@ export const useAuthenticationService = () => {
     const register = useCallback(
         async (registerDto: RegisterDto): Promise<string> => {
             try {
-                const response = await authenticationApi.apiAuthenticationRegisterPost(registerDto);
+                const response = await authenticationApi.register(registerDto);
```
Comment (Smightym8) [inline src/hooks/services/useAuthenticationService.ts:22]: Looks much better with the method names from open api :)

**Review by Smightym8 — APPROVED** [review]

File: src/App.tsx (hunk 1 above)
Comment (iiNomad23) [reply src/App.tsx:32]: Explicit type

File: src/hooks/services/useAuthenticationService.ts (hunk 2 above)
Comment (iiNomad23) [reply src/hooks/services/useAuthenticationService.ts:22]: Yes!

**Pull request merged**